aospy/data_loader.py

```diff
@@ -450,49 +473,44 @@ class GFDLDataLoader(DataLoader):
                  data_start_date=None, data_end_date=None,
-                 upcast_float32=None,
-                 preprocess_func=lambda ds, **kwargs: ds):
+                 upcast_float32=None, data_vars=None, coords=None,
+                 preprocess_func=None):
         """Create a new GFDLDataLoader"""
-        attrs = ['data_direc', 'data_dur', 'data_start_date', 'data_end_date',
-                 'preprocess_func']
         if template:
-            for attr in attrs:
-                setattr(self, attr, getattr(template, attr))
-
-            if data_direc is not None:
-                self.data_direc = data_direc
-            else:
-                self.data_direc = template.data_direc
-            if data_dur is not None:
-                self.data_dur = data_dur
-            else:
-                self.data_dur = template.data_dur
-            if data_start_date is not None:
-                self.data_start_date = data_start_date
-            else:
-                self.data_start_date = template.data_start_date
-            if data_end_date is not None:
-                self.data_end_date = data_end_date
-            else:
-                self.data_end_date = template.data_end_date
-            if preprocess_func is not None:
-                self.preprocess_func = preprocess_func
-            else:
-                self.preprocess_func = template.preprocess_func
-            if upcast_float32 is not None:
-                self.upcast_float32 = upcast_float32
-            else:
-                self.upcast_float32 = template.upcast_float32
+            self._maybe_inherit(template, 'data_direc', data_direc)
+            self._maybe_inherit(template, 'data_dur', data_dur)
+            self._maybe_inherit(template, 'data_start_date', data_start_date)
+            self._maybe_inherit(template, 'data_end_date', data_end_date)
+            self._maybe_inherit(template, 'upcast_float32', upcast_float32)
+            self._maybe_inherit(template, 'data_vars', data_vars)
+            self._maybe_inherit(template, 'coords', coords)
+            self._maybe_inherit(template, 'preprocess_func', preprocess_func)
         else:
             self.data_direc = data_direc
             self.data_dur = data_dur
             self.data_start_date = data_start_date
             self.data_end_date = data_end_date
-            self.preprocess_func = preprocess_func
-            if upcast_float32 is None:
-                self.upcast_float32 = True
-            else:
-                self.upcast_float32 = upcast_float32
+            self._set_default('upcast_float32', upcast_float32, True)
+            self._set_default('data_vars', data_vars, 'minimal')
+            self._set_default('coords', coords, 'minimal')
+            self._set_default('preprocess_func', preprocess_func,
+                              lambda ds, **kwargs: ds)
+
+    def _maybe_inherit(self, template, attr, value):
```

Comment:
Since this and `_set_default` aren't using any of the object's methods or attributes, I think it would be cleaner to define these at the module level as functions: `def _maybe_inherit(obj, template, attr, value):` etc.

Also, are the two different enough to warrant separate definitions?  Am I correct that `_set_default(obj, attr, value, getattr(template, attr)) == _maybe_inherit(obj, template, attr, value)`?

If you do keep both, I'd suggest making the signatures more homogenous: obj, attr, value, template/default.

Reply:
Indeed!  Good catch.  I'll clean these up when I get a chance.